plugins/by-name/none-ls/prettierd.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, config, ... }:
+let
+  cfg = config.plugins.none-ls.sources.formatting.prettierd;
+  ts-ls-cfg = config.plugins.lsp.servers.ts_ls;
+in
+{
+  options.plugins.none-ls.sources.formatting.prettierd = {
+    disableTsServerFormatter = lib.mkOption {
+      type = with lib.types; nullOr bool;
+      description = ''
+        Disables the formatting capability of the `ts_ls` language server if it is enabled.
+      '';
+      default = null;
+      example = true;
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    warnings = lib.optional ((cfg.disableTsServerFormatter == null) && ts-ls-cfg.enable) ''
+      You have enabled the `prettierd` formatter in none-ls.
+      You have also enabled the `ts_ls` language server which also brings a formatting feature.
+
+      - To disable the formatter built-in the `ts_ls` language server, set
+        `plugins.none-ls.sources.formatting.prettierd.disableTsServerFormatter` to `true`.
+      - Else, to silence this warning, explicitly set the option to `false`.
+    '';
+
+    plugins.lsp.servers.ts_ls =
+      lib.mkIf
+        (
+          cfg.enable
+          && ts-ls-cfg.enable
+          && (lib.isBool cfg.disableTsServerFormatter)
```

Comment:
Note: the parentheses aren't needed here, since function application takes precedent over most other operations, including `&&` operators.

However if we dropped the `nullOr` type then this line wouldn't be needed at all :grin:

Reply:
Done!